internal/objectstore/remote/retriever.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package remote
+
+import (
+	"context"
+	"io"
+
+	"github.com/juju/clock"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/s3client"
+	"github.com/juju/juju/internal/worker/apiremotecaller"
+)
+
+const (
+	// NoRemoteConnections is returned when there are no remote connections
+	// available.
+	NoRemoteConnections = errors.ConstError("no remote connections available")
+
+	// NoRemoteConnection is returned when there is no remote connection
+	// available.
+	NoRemoteConnection = errors.ConstError("no remote connection available")
```

Comment:
I recommend that semantically different errors are more dissimilar.

Reply:
Agree, I'll change.